src/main/java/com/rarchives/ripme/ripper/VideoRipper.java

```diff
@@ -44,19 +47,22 @@ public boolean addURLToDownload(URL url, File saveAs) {
         if (Utils.getConfigBoolean("urls_only.save", false)) {
             // Output URL to file
             String urlFile = this.workingDir + File.separator + "urls.txt";
+
+            FileWriter fw = null;
             try {
-                FileWriter fw = new FileWriter(urlFile, true);
+                fw = new FileWriter(urlFile, true);
                 fw.write(url.toExternalForm());
                 fw.write("\n");
-                fw.close();
+
                 RipStatusMessage msg = new RipStatusMessage(STATUS.DOWNLOAD_COMPLETE, urlFile);
                 observer.update(this, msg);
             } catch (IOException e) {
                 logger.error("Error while writing to " + urlFile, e);
                 return false;
+            } finally {
+                close(fw);
```

Comment:
If i add fw.close in the finally block, i will have to add or try/cacth around it, but i believe that try/catch inside the finally block is bad practice, so i created the close method, which contains the necessary try/catch to run fw.close